src/core/HatAllocator.ts

```diff
@@ -37,6 +37,10 @@ export class HatAllocator {
         this.toggleDecorations
       ),
 
+      // An event that fires when a text document opens
+      vscode.workspace.onDidOpenTextDocument(this.addDecorationsDebounced),
+      // An event that fires when a text document closes
+      vscode.workspace.onDidCloseTextDocument(this.addDecorationsDebounced),
```

Comment:
Before this change, we didn't retokenize a document when we changed the language to css.  VSCode treats a language change as closing and re-opening the editor, so we need to listen to these events to catch that

Reply:
Can we just listen to editor open? Adding decorations to an editor that has "closed" sounds like a weird thing to do :sweat_smile:

Reply:
It doesn't add decorations to the closed editor, but I think probably closing an editor could affect hats in other editors 🤷‍♂️.  It's debounced as well so shouldn't make much difference